app/src/main/java/ch/epfl/sweng/SDP/game/RankingFragment.java

```diff
@@ -92,35 +153,123 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
     }
 
+    private void updateUserStats(int starIncrease, int trophiesIncrease) {
+        Account account = Account.getInstance(getActivity()
+                .getApplicationContext());
+        account.changeStars(starIncrease);
+        account.changeTrophies(trophiesIncrease);
+        account.changeAverageRating(starIncrease);
+        account.increaseTotalMatches();
+    }
+
+    // TODO: Implement
+    private void updateScoresToCurrentUser(final DataSnapshot data) {
+
+        currentUserRef.runTransaction(new Transaction.Handler() {
+            @Override
+            public Transaction.Result doTransaction(final MutableData currentData) {
+                if (currentData.getValue() == null) {
+                    currentData.setValue(0);
+                } else {
+                    currentData.setValue((Long) currentData.getValue() + (Long) data.getValue());
+                }
+
+                return Transaction.success(currentData);
+            }
+
+            @Override
+            public void onComplete(@Nullable DatabaseError databaseError, boolean b, @Nullable DataSnapshot dataSnapshot) {
+                if (databaseError != null) {
+                    System.out.println("Firebase counter increment failed.");
+                } else {
+                    System.out.println("Firebase counter increment succeeded.");
+                }
+            }
+
+
+        });
+        // Set user stars to stars + stars
+        // Set trophies to trophies + position*10-20
+    }
+
     private void setFinishedCollectingRanking() {
-        finishedRef.child(Account.getInstance(getActivity()
-                .getApplicationContext()).getUsername()).setValue(1);
+        finishedRef.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                Integer value = dataSnapshot.getValue(Integer.class);
+                if (value != null) {
+                    finishedRef.setValue(++value);
+                }
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError databaseError) {
+
+            }
+        });
     }
 
     private class RankingAdapter extends ArrayAdapter<String> {
 
         private final String[] players;
+        private final Integer[] rankings;
+        private final Integer[] trophies;
+        private final Bitmap[] drawings;
+
 
-        private RankingAdapter(Context context, String[] players) {
+        private RankingAdapter(Context context, String[] players, Integer[] rankings, Integer[] trophies, Bitmap[] drawings) {
             super(context, 0, players);
             this.players = players;
+            this.rankings = rankings;
+            this.trophies = trophies;
+            this.drawings = drawings;
         }
 
         @NonNull
         @Override
         public View getView(int position, View convertView, @NonNull ViewGroup parent) {
```

Comment:
As far as I understood, according to the MVC pattern, this function should be moved into a separate class, right?

Reply:
I believe that this is a android specific "Adapter" like pattern that it used instead of MVC. This should be exactly the same pattern as described in the android documentation

Reply:
Oh I see, you are right.